lib/rucio/rse/protocols/gfal.py

```diff
@@ -166,7 +166,7 @@ def connect(self):
         # self.__ctx.set_opt_string("X509", "CERT", proxy)
         # self.__ctx.set_opt_string("X509", "KEY", proxy)
         self.__ctx.set_opt_string_list("SRM PLUGIN", "TURL_PROTOCOLS", ["gsiftp", "rfio", "gsidcap", "dcap", "kdcap"])
-        self.__ctx.set_opt_string("XROOTD PLUGIN", "XRD.WANTPROT", "gsi")
+        self.__ctx.set_opt_string("XROOTD PLUGIN", "XRD.WANTPROT", "gsi,unix")
```

Comment:
oh wow!

just a single comment: please add yourself as an author at the top of the file ;-)

Reply:
Amended!

Reply:
Shouldn't it be :
self.__ctx.set_opt_string("XROOTD PLUGIN", "XRD.WANTPROT", ["gsi", "unix"])
?

Reply:
Depends how the gfal context wants it. If it's set_opt_string then it has to be "gsi,unix", if it's set_opt_string_list, then it should be ["gsi","unix"]

I hope @nikmagini tried this and this is the correct variant ;-)

Reply:
My point is that I tried the fix manually on the functional test node, but it still fails :
2018-01-17 14:30:23,388 24426   INFO    Event triggered: BOTH GFAL2:CORE:COPY LIST:ENTER 
2018-01-17 14:30:23,388 24426   INFO    Event triggered: BOTH GFAL2:CORE:COPY LIST:ITEM file:///tmp/tmp6UxcgM/ESD.115ba8e29b71460cbca47eef9959e39c => root://eosatlas.cern.ch:1094//eos/atlas/atlasdatadisk/ruciotest/rucio/tests/5c/28/ESD.115ba8e29b71460cbca47eef9959e39c.rucio.upload
2018-01-17 14:30:23,388 24426   INFO    Event triggered: BOTH GFAL2:CORE:COPY LIST:EXIT 
2018-01-17 14:30:23,389 24426   INFO    Event triggered: BOTH xroot TRANSFER:ENTER file://///tmp/tmp6UxcgM/ESD.115ba8e29b71460cbca47eef9959e39c?xrd.gsiusrpxy=/opt/rucio/etc/ddmusr01.rfc.proxy => root://eosatlas.cern.ch:1094///eos/atlas/atlasdatadisk/ruciotest/rucio/tests/5c/28/ESD.115ba8e29b71460cbca47eef9959e39c.rucio.upload?xrd.gsiusrpxy=/opt/rucio/etc/ddmusr01.rfc.proxy
2018-01-17 14:30:23,390 24426   INFO    Event triggered: BOTH xroot TRANSFER:EXIT Job finished, [ERROR] Server responded with an error: [3010] Unable to open file /eos/atlas/atlasdatadisk/ruciotest/rucio/tests/5c/28/ESD.115ba8e29b71460cbca47eef9959e39c.rucio.upload; Operation not permitted

Nicolo, did you managed to make it work ?

Reply:
can you try on your testnode with set_opt_string_list instead of set_opt_string?

Reply:
Hi all,

1) I used set_opt_string("XROOTD PLUGIN", "XRD.WANTPROT", "gsi,unix") as recommended by gfal devs. I also tested this personally on my dev machine and it works.
In alternative, I also tested set_opt_string_list("XROOTD PLUGIN","XRD.WANTPROT",['gsi','unix']) and it also works; I think that the two syntaxes are equivalent.

2) As mentioned in a previous comment, however,  this is not expected to work elsewhere yet, due to a bug in the release gfal version. On my dev machine I installed this version of gfal from the testing repo to make it work:
gfal2-2.15.0-r1801151621

It's in the unstable repo here:
https://dmc-docs.web.cern.ch/dmc-docs/repositories.html